frontend/simple/views/containers/contributions/IncomeForm.vue

```diff
@@ -0,0 +1,245 @@
+                <i18n>will ensure you meet the mincome</i18n>
+              </template>
+            </input-amount>
+
+            <input-amount v-else-if="hasMinIncome" key="pledge"
+              label="How much do you want to pledge?"
+              :error="hasMinIncomeError"
+              @input="verifyInputPledge"
+              :value="this.form.pledge"
+              :placeholder="L('amout')"
+            >
+              <template slot="help">
+                <i18n tag="p">Define up to how much you pledge to contribute to the group each month.</i18n>
+                <i18n tag="p" class="c-help-extra">You can pledge any amount (even $1 million!) Only the minimum needed amount will be given.</i18n>
+              </template>
+            </input-amount>
+          </fieldset>
+          <fieldset class="fieldset" v-if="!!$v.form.option.$model">
+            <payment-methods />
+          </fieldset>
+        </div>
+
+        <group-pledges-graph class="column c-pledgesGraph"
+          :userPledgeAmount="userPledgeAmount"
+          :userIncomeAmount="userIncomeAmount"
+        />
+      </div>
+
+      <div class="field is-grouped">
+        <p class="control">
+          <i18n tag="button" class="button is-primary" type="submit">{{saveButtonText}}</i18n>
+        </p>
+        <p class="control">
+          <i18n tag="button" class="button" type="button" @click="cancelForm">Cancel</i18n>
+        </p>
+      </div>
+      </form>
+  </modal>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-wrapper {
+  width: 50rem;
+  max-width: 100%;
+  height: 100%;
+  overflow: auto;
+
+  @include mobile {
+    height: 100vh;
+  }
+
+  &-columns {
+    @include mobile {
+      flex-direction: column;
+    }
+  }
+}
+
+.c-help-extra {
+  margin-top: $gi-spacer-sm;
+}
+
+.c-form,
+.c-pledgesGraph {
+  flex-basis: 50%;
+}
+
+.c-pledgesGraph {
+  align-items: flex-start;
+  flex-grow: 1;
+
+  @include mobile {
+    order: -1; // display on the top for better understanding on small screens
+    padding: $gi-spacer $gi-spacer-sm;
+  }
+
+  @include tablet {
+    justify-content: center;
+    margin-top: -$gi-spacer-lg;
+  }
+}
+
+</style>
+<script>
+import { validationMixin } from 'vuelidate'
+import { requiredIf, required, maxValue } from 'vuelidate/lib/validators'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Modal from '../../components/Modal/ModalBasic.vue'
+import TextWho from '../../components/TextWho.vue'
+import { debounce } from '../../../utils/giLodash.js'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+
+export default {
+  name: 'IncomeForm',
+  mixins: [ validationMixin ],
+  components: {
+    Modal,
+    InputAmount,
+    TextWho,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  props: {
+    isEditing: Boolean
+  },
+  data () {
+    return {
+      ephemeral: {
+        formVerified: false
```

Comment:
It's not clear to me that `formVerified` is needed? Can't you simply check `$v.$invalid` or `$v.form.$error`?

Reply:
I've updated the code based on your suggestions during our last call, thank you! 🙌